### PR TITLE
Add C++ gRPC client to the Oak Functions integration test

### PR DESCRIPTION
**Pull request by ipetr0v**
This PR makes Oak Functions integration test to run C++ gRPC client CLI.

Fixes https://github.com/project-oak/oak/issues/4069

### CODE REVIEW

**Review by tiziano88 — APPROVED** [review]

File: cc/transport/grpc_streaming_transport.h [hunk 1]
```diff
@@ -96,18 +96,16 @@ class GrpcStreamingTransport : public TransportWrapper {
...
-    if (status.ok()) {
-      return response;
-    } else {
-      return absl::InternalError("couldn't send request: " + status.error_message());
+    if (!channel_reader_writer_->Read(&response)) {
+      return absl::InternalError("couldn't receive response");
     }
+    return response;
   }
 };
```
Comment (pmcgrath17) [inline cc/transport/grpc_streaming_transport.h:110]: Please add a deconstructor to close the stream.

Comment (ipetr0v) [reply cc/transport/grpc_streaming_transport.h]: Done

**Comment by pmcgrath17** [comment]
Did this work as expected in tests?

**Comment by ipetr0v** [comment]
> Did this work as expected in tests?

Yes, the tests are working:
```
I0000 00:00:1689344716.397394 1764595 grpc_client_cli.cc:54] connecting to: localhost:23261
I0000 00:00:1689344716.401647 1764595 grpc_client_cli.cc:63] creating Oak Client
I0000 00:00:1689344716.402468 1764595 grpc_client_cli.cc:74] sending request: {"lat":0,"lng":0}
I0000 00:00:1689344716.437730 1764595 grpc_client_cli.cc:80] received response: {"temperature_degrees_celsius":29}
```

**Pull request merged**